### PR TITLE
[CP-2398] Added modal and improved buttons

### PR DESCRIPTION
**Pull request by mkurczewski**
JIRA Reference: [CP-2398]

### :memo: Description ️
- Added Modal component
- Added plain text and markdown text components
- Added modals demo on "Demo" page
- Improved fonts rendering

<details>
<summary>Screenshots</summary>

image

image

image

</details>

### :muscle: Checklist before requesting a review - nice to have

- getState function in async thunk actions is correctly typed
- redux selectors are used in components / prop drilling is reduce

### :exclamation: Checklist before merging a pull request

- change went through the QA process
- translations are updated in dedicated application
- [CHANGELOG.md](./CHANGELOG.md) is updated

[CP-2398]: https://appnroll.atlassian.net/browse/CP-2398?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/generic-view/store/src/lib/modals/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  closeDomainModals,
+  closeModal,
+  openModal,
+  replaceModal,
+} from "./actions"
+
+export interface Modal {
+  key: string
+  domain?: string
+  replaced?: boolean
```
Comment (mkurczewski) [inline libs/generic-view/store/src/lib/modals/reducer.ts:18]: This prop helps in situation when modal replaces another one, and then the new one is closed, so the previous (with replaced: true) can be shown again.

File: libs/generic-view/ui/src/lib/data-rows/text-formatted.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  content: string
+}
+
+const TextFormatted: APIFC<Data> = ({ data }) => {
+  return <Content>{data?.content}</Content>
+}
+
+export default withData(TextFormatted)
+
+const Content = styled(Markdown)`
```
Comment (mkurczewski) [inline libs/generic-view/ui/src/lib/data-rows/text-formatted.tsx:21]: Below styles will be improved in future, but we need to get final designs first (in progress).

File: libs/generic-view/ui/src/lib/buttons/button-base.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { useDispatch } from "react-redux"
+import { useHistory } from "react-router"
+import { ButtonAction, useScreenTitle } from "generic-view/utils"
+
+interface Props extends HTMLAttributes<HTMLButtonElement> {
+  action: ButtonAction
+  viewKey?: string
+}
+
+export const ButtonBase: FunctionComponent<Props> = ({ action, ...props }) => {
```
Comment (dkarski) [inline libs/generic-view/ui/src/lib/buttons/button-base.tsx:24]: 🟢 
The "Base Button" in the UI library serves as a raw button component, providing minimal complexity and independence for easy integration across various parts of the project.

Dependency on Context:
The component heavily relies on `useDispatch` and `useHistory`, potentially posing challenges for testing and maintenance. Suggestion: Consider separating these dependencies for increased flexibility.

Lack of Action Translation:
Action texts, like `open-modal` or `close-modal`, are directly embedded in the code, making interface localization challenging. A recommended consideration is to create an enum or utilize external resources to store these texts. This approach would significantly ease translation management for diverse languages.

Missing `onClick` Handling:
The absence of direct support for onClick limits the component's flexibility in user interactions. Suggestion: Adding `onClick` support would enable more diverse usage in different contexts.

File: libs/generic-view/ui/src/lib/buttons/buttons.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import ButtonText from "./button-text"
+import ButtonLink from "./button-link"
+
+export const buttons = {
```
Comment (dkarski) [inline libs/generic-view/ui/src/lib/buttons/buttons.ts:9]: ⚪ Neutral, Suggestion:

Continuing the discussion from the previous section https://github.com/mudita/mudita-center/pull/1667/files#r1457039056

```tsx
/**
 * The `Button` component handles various variants, such as text, link, icon, or even a custom button.
 * This serves as a compromise between a generic button component and a more complex parent component that manages navigation and modal usage.
 * Variants allow flexible customization of the button's appearance and functionality for specific use cases while preserving overall control in the parent component.
 */
interface ButtonProps extends BaseButtonProps {
  type: 'default' | 'primary' | 'secondary';
  theme: Theme;
  variant: 'text' | 'link' | 'custom'; // Add new variants
  action: ButtonAction;
}

const Button: FunctionComponent<ButtonProps> = ({
  theme,
  type,
  variant,
  action,
  ...restProps
}) => {
  // ... (Remainder of the component unchanged)

  // Depending on the variant, render the appropriate component
  switch (variant) {
    case 'text':
      return <ButtonText {...props} onClick={handleButtonClick} />;
    case 'link':
      return <ButtonLink {...props} onClick={handleButtonClick} />;
    case 'icon':
      return <BaseButtonWithIcon {...props} onClick={handleButtonClick} />;
    case 'custom':
      // Add handling for a custom variant if necessary
      return <CustomButton {...props} onClick={handleButtonClick} />;
    default:
      return <BaseButton {...props} onClick={handleButtonClick} />;
  }
};
```
This component has been adapted to accommodate diverse variants, offering a balance between a generic button component and a more intricate parent component responsible for navigation and modal usage. Variants enable adaptable customization of the button's visual and functional aspects for specific use cases, all while maintaining overarching control in the parent component.

**Review by dkarski — APPROVED** [review]
✅ 

The PR is well-prepared, clean code, no objections. Comments are philosophical, please don't take them personally. Open to discussion. Not blocking the PR, strong approval from my side.

**Comment by patryk-sierzega** [comment]
Task in Jira doesn't have any description, any acceptance criteria, etc. I think we should avoid this situations.

**Review by lkowalczyk87 — APPROVED** [review]

**Pull request merged**